src/model/channel/attachment_type.rs

```diff
@@ -107,11 +127,11 @@ impl<'a> From<&'a PathBuf> for AttachmentType<'a> {
     }
 }
 
-impl<'a> From<(&'a File, &str)> for AttachmentType<'a> {
-    fn from(f: (&'a File, &str)) -> AttachmentType<'a> {
+impl<'a> From<(&'a File, Cow<'static, str>)> for AttachmentType<'a> {
+    fn from((file, filename): (&'a File, Cow<'static, str>)) -> AttachmentType<'a> {
         AttachmentType::File {
-            file: f.0,
-            filename: f.1.to_string(),
+            file,
+            filename,
         }
     }
 }
```

Comment:
Github won't let me suggest changes directly here, but you can change this to take an `Into<Cow>` as follows:

```rust
impl<'a, S> From<(&'a File, S)> for AttachmentType<'a>
where
    S: Into<Cow<'static, str>>,
{
    fn from((file, filename): (&'a File, S)) -> AttachmentType<'a> {
        AttachmentType::File {
            file,
            filename: filename.into(),
        }
    }
}
```